### PR TITLE
Fix web api HTTP to be 9600 as Default

### LINKED ISSUE #4648: http server default port
Change the default port of the http server, the current default is 9292. This port conflict with ES's range.
@suyograo  suggested we move to 9600, a quick google search doesnt show any popular application using that port.

Also we should be able to change it when starting logstash with a `cli` command, 
`--http-port` this can be moved to the settings file later to `http.port` setting.

### PR DESCRIPTION
**Pull request by purbon**
Fixes #4648

### CODE REVIEW

**Comment by purbon** [comment]
@ph @jsvd thoughts and reviews more than welcome.

File: logstash-core/lib/logstash/webserver.rb [hunk 1]
```diff
@@ -17,11 +17,14 @@ class WebServer
...
       @logger      = logger
       @options     = {}
-      @cli_options = options.merge({ :rackup => ::File.join(::File.dirname(__FILE__), "api", "init.ru")  })
+      @cli_options = options.merge({ :rackup => ::File.join(::File.dirname(__FILE__), "api", "init.ru"), 
+                                     :binds => ["tcp://0.0.0.0:#{options[:http_port]}"] })
       @status      = nil
 
       parse_options
 
+      puts @options
```
Comment (ph) [inline logstash-core/lib/logstash/webserver.rb]: Debugging?

File: logstash-core/lib/logstash/agent.rb [hunk 2]
```diff
@@ -34,6 +34,8 @@ def initialize(params)
     @pipelines = {}
     @started_at = Time.now
     @node_name = params[:node_name] || Socket.gethostname
+    @web_api_http_port = params[:web_api_http_port] || 9600
```
Comment (ph) [inline logstash-core/lib/logstash/agent.rb]: I think the default value should be handled inside the webservice class, It's not a concern of the agent to setup the default values for the port. Also moving it there align with our discussion from yesterday concerning a runner class.

File: logstash-core/lib/logstash/webserver.rb (hunk 1 above)
Comment (purbon) [inline logstash-core/lib/logstash/webserver.rb]: crap yes.

File: logstash-core/lib/logstash/agent.rb (hunk 2 above)
Comment (purbon) [inline logstash-core/lib/logstash/agent.rb]: ++

**Comment by purbon** [comment]
@ph comments amended, thanks a lot for your time.

**Comment by purbon** [comment]
please jenkins, test this.

**Comment by ph** [comment]
Do you think we should add a test for this? Maybe a unit test against the webserver?

**Comment by purbon** [comment]
we should for sure, but unit test this would be a bit complicated as it
will need to mock the web app somehow.... can we open an issue to add this
in next iteration? will need to think about the right way to test it.

- purbon

**Comment by ph** [comment]
+1 create an issue1 lgtm

**Pull request closed without merging**